packages/pn-pa-webapp/src/components/Notifications/NotificationSettingsDrawer.tsx

```diff
@@ -0,0 +1,181 @@
+import { useFormik } from 'formik';
+import { useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import * as yup from 'yup';
+
+import CloseIcon from '@mui/icons-material/Close';
+import {
+  Box,
+  Drawer,
+  FormControl,
+  FormControlLabel,
+  IconButton,
+  MenuItem,
+  Radio,
+  RadioGroup,
+  Typography,
+} from '@mui/material';
+import { CustomDropdown } from '@pagopa-pn/pn-commons';
+import { LANGUAGES } from '@pagopa-pn/pn-commons/src/utility/costants';
+import { ButtonNaked, LangCode } from '@pagopa/mui-italia';
+
+import { BILINGUALISM_LANGUAGES, NewNotificationLangOther } from '../../models/NewNotification';
+import { setAdditionalLanguages } from '../../redux/auth/actions';
+import { useAppDispatch, useAppSelector } from '../../redux/hooks';
+import { RootState } from '../../redux/store';
+
+const NotificationSettingsDrawer = () => {
+  const dispatch = useAppDispatch();
+  const [openDrawer, setOpenDrawer] = useState(false);
+  const { t, i18n } = useTranslation(['notifiche']);
+  const additionalLanguages = useAppSelector(
+    (state: RootState) => state.userState.additionalLanguages
+  );
+
+  const toggleDrawer = () => setOpenDrawer(!openDrawer);
+
+  const onCloseDrawer = async () => {
+    await formik.validateForm();
+    if (!formik.isValid) {
+      await formik.setTouched({ lang: true, additionalLang: true });
+      return;
+    }
+    formik.handleSubmit();
+    setOpenDrawer(false);
+  };
+
+  const validationSchema = yup.object({
+    lang: yup.string().oneOf(['it', NewNotificationLangOther]).required(),
+    additionalLang: yup.string().when('lang', {
+      is: NewNotificationLangOther,
+      then: yup.string().required(t('settings.select-language-to-confirm')),
+    }),
+  });
+
+  const initialValues = {
+    lang: 'it',
+    additionalLang: '',
+  };
+  const formik = useFormik({
+    initialValues,
+    validationSchema,
+    onSubmit: (values) => {
+      void dispatch(
+        setAdditionalLanguages(
+          values.lang === NewNotificationLangOther ? [values.additionalLang] : []
+        )
+      );
+    },
+  });
+
+  const languages = useMemo(() => {
+    const currentLang = i18n.language?.substring(0, 2) as LangCode;
+    return LANGUAGES[currentLang] ?? LANGUAGES.it;
+  }, [i18n.language]);
+
+  const initForm = async () => {
+    await formik.setValues(
+      additionalLanguages && additionalLanguages.length > 0
+        ? {
+            lang: NewNotificationLangOther,
+            additionalLang: additionalLanguages[0],
+          }
+        : initialValues
+    );
+  };
+
+  useEffect(() => {
+    if (openDrawer) {
+      void initForm();
+    }
+  }, [openDrawer, additionalLanguages]);
```

Comment:
Remove this part and use `initialValues` instead. Also add `enableReinitialize: true` on `useFormik`